tests/test_driver.sh

```diff
@@ -0,0 +1,113 @@
+    XDG_DATA_HOME="$homedir/xdg_data_home"
+    export XDG_DATA_HOME
+    mkdir -p $XDG_DATA_HOME/fish || die
+
+    # cp -a ../test/xdg_config_home "$homedir/"
+    XDG_CONFIG_HOME="$homedir/xdg_config_home"
+    export XDG_CONFIG_HOME
+    mkdir -p $XDG_CONFIG_HOME/fish || die
+
+    # These are used read-only so it's OK to symlink instead of copy
+    rm -f "$XDG_CONFIG_HOME/fish/functions"
+    ln -s "$PWD/test_functions" "$XDG_CONFIG_HOME/fish/functions" || die "Failed to symlink"
+
+    # Set the function path at startup, referencing the default fish functions and the test-specific
+    # functions.
+    fish_init_cmd="set fish_function_path ${XDG_CONFIG_HOME}/fish/functions ${BUILD_ROOT}/share/functions"
+
+    __fish_is_running_tests="$homedir"
+    export __fish_is_running_tests
+
+    # Set locale information for consistent tests. Fish should work with a lot of locales but the
+    # tests assume an english UTF-8 locale unless they explicitly override this default. We do not
+    # want the users locale to affect the tests since they might, for example, change the wording of
+    # logged messages.
+    #
+    # TODO: set LANG to en_US.UTF-8 so we test the locale message conversions (i.e., gettext).
+    unset LANGUAGE
+    # Remove "LC_" env vars from the test environment
+    for kv in $(env); do
+        if echo "$kv" | grep -Eq "^LC_"; then
+            unset "$(echo "$kv" | grep -oE "^[^=]+")"
+        fi
+    done
+    # Set the desired lang/locale tests are hard-coded against
+    export LANG="C"
+    export LC_CTYPE="en_US.UTF-8"
+
+    # These env vars should not be inherited from the user environment because they can affect the
+    # behavior of the tests. So either remove them or set them to a known value.
+    # See also tests/interactive.fish.
+    export TERM=xterm
+    unset COLORTERM
+    unset INSIDE_EMACS
+    unset ITERM_PROFILE
+    unset KONSOLE_PROFILE_NAME
+    unset KONSOLE_VERSION
+    unset PANTHEON_TERMINAL_ID
+    unset TERM_PROGRAM
+    unset TERM_PROGRAM_VERSION
+    unset VTE_VERSION
+    unset WT_PROFILE_ID
+    unset XTERM_VERSION
+
+    # Set a marker to indicate whether colored output should be suppressed
+    suppress_color=""
+    if ! tty 0>&1 > /dev/null; then
+        suppress_color="yes"
+    fi
+    export suppress_color
+
+    # Source test util functions at startup
+    fish_init_cmd="${fish_init_cmd} && source ${TESTS_ROOT}/test_util.fish";
+
+    # Run the test script, but don't exec so we can do cleanup after it succeeds/fails
+    # echo $PWD
+    # echo "BUILD_ROOT: $BUILD_ROOT"
+    # echo "TESTS_ROOT: $TESTS_ROOT"
+    env HOME=$homedir "${BUILD_ROOT}/test/root/bin/fish" \
+        --init-command "${fish_init_cmd}" \
+        "$fish_script" "$script_args"
+    test_status="$?"
+
+    echo test completed in $homedir with status $test_status
+
+    rm -rf "$homedir"
+    exit "$test_status"
+else
+    echo "Recursive test invocation detected!" 1>&2
+    exit 10
```

Comment:
> I noticed is that "unsupported" littlecheck tests are listed as "passed" with ctest while littlecheck would print
> 
> ![image](https://user-images.githubusercontent.com/6853872/112591778-30d24780-8e05-11eb-9181-a330d144cda2.png)
> 
> Of course that's a really minor detail since we only have like 2 tests with `REQUIRES`.

If little check returns a specific error code, we can program that to be a “skipped” marker and have ctest act accordingly. I’m guessing it’s just returning 0 here.

Reply:
Done.